### PR TITLE
Create image for blackbox monitoring of jupyterhub-db

### PR DESCRIPTION
**Pull request by anishasthana**
Signed-off-by: Anish Asthana <anishasthana1@gmail.com>

### CODE REVIEW

File: monitoring/jupyterhub-db-probe/src/main.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
+import os
+import time
+
+from prometheus_client import start_http_server, Gauge
+from sqlalchemy import create_engine
+
+DATABASE_RESPONSE_TIME = Gauge('jupyterhub_db_response_time', 'Time taken for Jupyterhub DB to respond. Negative values indicate failures')
+
+def main():
+    user = os.getenv("JUPYTERHUB_DB_USER", "jupyterhub")
```
Comment (maulikjs) [inline monitoring/jupyterhub-db-probe/src/main.py:10]: should we keep defaults here?

File: monitoring/jupyterhub-db-probe/src/main.py [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    while True:
+        DATABASE_RESPONSE_TIME.set(connect_to_db(user, password, host, port))
+
+
+def connect_to_db(user, password, host, port):
+    try:
+        start_time = time.time_ns() // 1_000_000
+        db_string = f"postgresql://{user}:{password}@{host}:{port}/jupyterhub"
+        db = create_engine(db_string)
+        foo = db.execute("SELECT * FROM users")
```
Comment (maulikjs) [inline monitoring/jupyterhub-db-probe/src/main.py:26]: I'd say add a `LIMIT 1` to this query for consistency

File: monitoring/jupyterhub-db-probe/src/main.py (hunk 1 above)
Comment (anishasthana) [reply monitoring/jupyterhub-db-probe/src/main.py:10]: I like having the defaults cause there is some stuff in flux upstream for ODH and secrets. So while that stuff is being done we have something working. I'm happy to remove it once the secrets stuff is set in stone.

Comment (maulikjs) [reply monitoring/jupyterhub-db-probe/src/main.py:10]: sounds good 👍

**Review by maulikjs — APPROVED** [review]

**Pull request merged**